src/libsyntax_ext/proc_macro_registrar.rs

```diff
@@ -232,7 +257,8 @@ impl<'a> Visitor<'a> for CollectProcMacros<'a> {
         let mut found_attr: Option<&'a ast::Attribute> = None;
 
         for attr in &item.attrs {
-            if attr.check_name("proc_macro_derive") || attr.check_name("proc_macro_attribute") {
+            if attr.check_name("proc_macro_derive") || attr.check_name("proc_macro_attribute") ||
+                attr.check_name("proc_macro") {
```

Comment:
Eh, I don't like either of those so it's your call. I would almost prefer a `.check_names()` method which can take a slice or iterator or something.